user/filters.py

```diff
@@ -9,14 +11,16 @@ class UserFilter(filters.FilterSet):
     search = filters.CharFilter(field_name="search", method="filter_search")
 
     def filter_search(self, queryset: QuerySet, name: str, value: str) -> QuerySet:
-        user_by_first_name = Q(first_name__icontains=value)
-        user_by_last_name = Q(last_name__icontains=value)
+        name, last_name, *_ = chain(value.split(" ", 1), [""])
```

Comment:
https://docs.python.org/3/library/itertools.html#itertools.chain
> Used for treating consecutive sequences as a single sequence.

We are using a single sequence resulting from the split of a single string value with spaces. And we are leaving out people with big names like me 😢 :  "Ricardo Monteiro e Lima" If I searched for "Lima". A good approach is tokenizing the string and dynamically composing the query:
```python
query = Q()
for token in value.split(" "):
            query |= Q(first_name__icontains=token)
            query |= Q(last_name__icontains=token)
            query |= Q(google_drive_spreadsheet_id=token)

result_queryset = queryset.filter(query)
```